### DISCUSSION TITLE
Expert Mode For Frequently Used CLI Commands

### DISCUSSION THREAD

**Discussion by robmulally** [iNavFlight/inav-configurator#1412] [Ideas]
**Disclaimer:** 
This might be my first GitHub comment for INAV so i hope this is in the right place. I did some searching and Didn't find it previously mentioned/discussed.

**TLDR:** I think expert mode toggle switch similar to BF might be a nice option for frequently used CLI commands rather than hiding them, for sensitive elements a Color code system could be used to know if modifying something is safe or not.

**Background:**
I recently started using INAV as a Legal requirement in Australia to ensure all my FPV Drones are compatible with Drone Operating Manual. I'm really enjoying the experience and have been watching as many youtube video about Inav as possible and am excited about the upcoming release and was watching this video: https://www.youtube.com/watch?v=vIAtb1iv0wk&t=1669s

**IDEA A&B:** 
As per the Video above the idea of removing more and more UX elements is a great idea!

A) For those UX elements left a colour code system could be used to warn users of playing with them is safe or not and perhaps they are greyed out or unchangable unless B is enabled. (see below)

B) B is the idea to use a Expert mode like BF to enable use of the "RED-Warning" Elements Visible as well as show other elements that are frequently tweaked  

**My Motivation:**
At the moment watching "Guides" on INAV i have to goto the CLI a lot, and excuse me if some of these things are added but if in an Official INAV video the Presenter has to use the CLI, that perhaps should be a Flag to say that it should be in GUI and either Colour Coded A, in Expert Mode B or both. 

** Examples **
Here are three things that I had to read through line by line of the DUMP to find that was very useful for me as a BF convert.
`# Disarm Delay
set switch_disarm_delay = 0

# Navigation
# Allow Bypass for Take Off - Hold Throttle Down and Right 
set nav_extra_arming_safety = ALLOW_BYPASS
set nav_disarm_on_landing = ON
set nav_mc_auto_disarm_delay = 250

# Black Box Recording
set debug_mode = GYRO`

Thanks for reading and apologies if this is an idea already discussed and buried I just couldn't find it.

Thanks
Rob

**Comment by MrD-RC** [comment]
Just my 2p. The expert mode is all well and good. But, everyone just enables it anyway. When I was first playing about with FCs, I had a quad and BetaFlight. Even back then, not knowing what I was doing, I had expert mode on. Why? Because to do something, I had to enable it. And in the video I was following, Bardwell just said something along the lines of “Just enable expert mode, because why wouldn’t you”. 

I don’t really think the expert mode does anything useful.

I’m really not knocking your idea. It is quicker to change things without having to use the CLI. But on the same token, the things you have listed could be part of a CLI setup script which you just run as part of the installation.

**Reply by robmulally** [reply]
No Problem at all, ideas should always be challenged!


This was a direct response to the you tube where he mentioned even more options be REMOVED to keep it simple as thats what "people" want. I can't imagine any INAV users wanting to keep it simple I would like to see everything that INAV can do.

Regarding Setup Script: 

Yes , I have a full script that I use and found via the DUMP file.

Also,
I'm currently working on an Excel Document that Gives me all the CLI values I need and also provides a Tuning Guide for myself to track PD PI and IP Ratios to mimic the sliders in BF. I just spent 5 hours on it and that is when I realized I'm just building a GUI replacement in Excel which prompted this suggestion.. 

A simple guideline should be that if your "setup" guide has users going into the CLI then you are missing UX elements. 

Rob.
